core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -861,9 +861,8 @@ impl<G: L1GasPriceProvider> EthNamespace<G> {
                     .map_err(|err| internal_error(METHOD_NAME, err))?;
                 *from_block = logs
```

Comment:
Off-topic (?): I don't quite like that in `TypedFilter::Blocks`, the cursor (named `from_block` in the code above) is exclusive (i.e., `get_block_hashes_after(from_block, _)` fetches block hashes starting from `from_block + 1`), but here, the cursor is inclusive. I think this could be the cause of the bug in the first place. IMO, it would make sense to uniformly use either exclusive or inclusive cursor for all `TypedFilter` variants (I'd slightly prefer inclusive, but really any consistent option would do fine).

Reply:
> MO, it would make sense to uniformly use either exclusive or inclusive cursor for all TypedFilter variants (I'd slightly prefer inclusive, but really any consistent option would do fine).

Yep